cpp/mrc/include/mrc/node/sink_properties.hpp

```diff
@@ -43,6 +43,13 @@ class NullReadableEdge : public edge::IEdgeReadable<T>
 
         return channel::Status::error;
     }
+
+    channel::Status await_read_until(T& t, const mrc::channel::time_point_t& tp) override
+    {
+        throw std::runtime_error("Attempting to read from a null edge. Ensure an edge was established for all sinks.");
```

Comment:
Do we want to throw or return an error status? Only one is strictly necessary.

In `source_properties.hpp` we only throw.

In `test_nodes.hpp` we do both.

Reply:
In `sink_properties.hpp` this is for the `NullReadableEdge` class which shouldn't allow reads of any sort. I was following the implementation of `await_read` for the `await_read_until` method. In this case I think the `throw` is the correct thing. I wasn't sure why the `await_read` also has a `return channel::Status::error;` I suspect it was just there to satisfy a lint check.